app/components/add-request.js

```diff
@@ -37,20 +38,24 @@ export default Ember.Component.extend({
       );
     },
 
-    removeRequest(reqId) {
-      var url = `/goodcity_requests/${reqId}`;
-      var req = this.get("store").peekRecord("goodcity_request", reqId);
-      var loadingView = getOwner(this)
-        .lookup("component:loading")
-        .append();
-      new AjaxPromise(url, "DELETE", this.get("session.authToken"))
-        .then(data => {
-          this.get("store").pushPayload(data);
-        })
-        .finally(() => {
-          loadingView.destroy();
-          this.get("store").unloadRecord(req);
-        });
+    async removeRequest(reqId) {
+      const req = this.get("store").peekRecord("goodcity_request", reqId);
+      this.runTask(await this.get("goodcityRequest").deleteRequest(reqId));
+      this.get("store").unloadRecord(req);
+    },
+
+    async assingPackageType(reqId) {
+      const pkgType = await this.get("packageService").getPackageType();
+
+      if (pkgType) {
+        this.runTask(
+          this.get("goodcityRequest").updateGcRequest(reqId, {
+            package_type_id: pkgType.get("id"),
+            quantity: 1,
+            order_id: this.get("order.id")
+          })
+        );
+      }
```

Comment:
I feel the name `goodcityRequest` is a bit misleading. Doesn't indicate that it's a service.

I'd also recommend renaming `getPackageType` to `userPickPackageType` to be consistent with the other methods that trigger UI overlays. Also more explicit

Reply:
Done!